tap-snapshots/test/lib/commands/dist-tag.js.test.cjs

```diff
@@ -20,7 +20,22 @@ latest: 1.0.0
 `
 
 exports[`test/lib/commands/dist-tag.js TAP ls on missing package > should log no dist-tag found msg 1`] = `
-dist-tag ls Couldn't get dist-tag data for foo@*
+ls Couldn't get dist-tag data for Result {
```

Comment:
This is a fun one.  We're now evaluating the raw data instead of the `toString` version.  I think ultimately we should still want to test what the actual string that makes it to the logfile will be?

Reply:
This is what is being shown to the user. Previously the tests were calling toString() on specific logged arguments. Now we are snapshoting exactly what users see.

Reply:
I'm relatively sure users will not see `Result {`, that object has a custom toString: https://github.com/npm/npm-package-arg/blob/17651118e122ea7c95a930285b228ccb1609652c/lib/npa.js#L168

Reply:
This is what I see with the current version of npm (not this branch):
```
❯ npm dist-tag ls -w smoke-tests
@npmcli/smoke-tests:
npm ERR! dist-tag ls Couldn't get dist-tag data for Result {
npm ERR! dist-tag ls   type: 'range',
npm ERR! dist-tag ls   registry: true,
npm ERR! dist-tag ls   where: undefined,
npm ERR! dist-tag ls   raw: '@npmcli/smoke-tests',
npm ERR! dist-tag ls   name: '@npmcli/smoke-tests',
npm ERR! dist-tag ls   escapedName: '@npmcli%2fsmoke-tests',
npm ERR! dist-tag ls   scope: '@npmcli',
npm ERR! dist-tag ls   rawSpec: '*',
npm ERR! dist-tag ls   saveSpec: null,
npm ERR! dist-tag ls   fetchSpec: '*',
npm ERR! dist-tag ls   gitRange: undefined,
npm ERR! dist-tag ls   gitCommittish: undefined,
npm ERR! dist-tag ls   gitSubdir: undefined,
npm ERR! dist-tag ls   hosted: undefined
npm ERR! dist-tag ls }
```

Reply:
*sigh* well that's a bug.  It's supposed to be stringifying it.

```js
> console.log(`beep boop ${require('npm-package-arg')('foo')}`)
beep boop foo@*
```

So the old snapshots weren't even showing us the bug.  We'll fix this later, out of scope for this PR.

Reply:
Still batting 1.000 on finding bugs while removing layers of abstraction from tests.